include/sas.h

```diff
@@ -114,42 +110,46 @@ class SAS
   typedef uint64_t TrailId;
   typedef uint64_t Timestamp;
 
-#if HAVE_ZLIB_H
-  // Compression-related classes are only available if zlib is
+  // Compression profile
   class Profile
   {
   public:
-    inline Profile(std::string dictionary) : _dictionary(dictionary) {}
+    enum CompressionType {
+      ZLIB = 0,
+      LZ4
+    };
+
+    inline Profile(std::string dictionary, CompressionType t = ZLIB) : _dictionary(dictionary), _type(t) {}
+    inline Profile(CompressionType t) : _dictionary(""), _type(t) {}
```

Comment:
I was going to say that you can use a delegating constructor here, i.e. `Profile(CompressionType t) : Profile("", t) {}`.  However, I think this is C++ 11 only, and I'm not sure whether our sas-client assumes that.  (Clearwater does.)